lib/pyraf/generic.py

```diff
@@ -457,8 +457,8 @@ def ambiguity(self, children):
             sortlist.append((len(rhs), rule))
             name2index[rule] = i
         sortlist.sort()
-        list = list(map(lambda (a, b): b, sortlist))
-        return children[name2index[self.resolve(list)]]
+        list_ = list(map(lambda a, b: b, sortlist))
```

Comment:
This is probably wrong; it should be as in [pysynphot/spark.py](https://github.com/spacetelescope/pysynphot/blob/7057c7ba4af7ffecb84d1e94f663639abf900f32/pysynphot/spark.py#L394-L395): 

```Python
alist = [s[1] for s in sortlist]
return children[name2index[self.resolve(alist)]]
```

Maybe copy the whole file from there.